packages/lit-auth-client/src/lib/providers/EthWalletProvider.ts

```diff
@@ -50,47 +54,92 @@ export default class EthWalletProvider extends BaseProvider {
+    });
+  }
+
+  /**
+   * Generate a wallet signature to use as an auth method
+   * 
+   * @param {EthWalletAuthenticateOptions} options
+   * @param {string} [options.address] - Address to sign with
+   * @param {function} [options.signMessage] - Function to sign message with
+   * @param {string} [options.chain] - Name of chain to use for signature
+   * @param {number} [options.expiration] - When the auth signature expires
+   * @returns {Promise<AuthMethod>} - Auth method object containing the auth signature
+   * @static
+   * @memberof EthWalletProvider
+   * 
+   * @example
+   * ```typescript
+   *   const authMethod = await EthWalletProvider.authenticate({
+   *      signer: wallet,
+   *      litNodeClient: client,
+   *   });
+   * ```
+   */
+  public static async authenticate({
+    signer,
+    address,
+    chain,
+    litNodeClient,
+    expiration,
+    domain,
+    origin,
+  }: {
+    signer: ethers.Signer | ethers.Wallet | any;
+    litNodeClient: LitNodeClient;
+    address?: string;
+    chain?: string;
+    expiration?: string;
+    domain?: string;
+    origin?: string;
+  }): Promise<AuthMethod> {
+    chain = chain || 'ethereum';
 
     let authSig: AuthSig;
 
-    if (address && signMessage) {
+    if (signer?.signMessage) {
       // Get chain ID or default to Ethereum mainnet
       const selectedChain = LIT_CHAINS[chain];
       const chainId = selectedChain?.chainId ? selectedChain.chainId : 1;
 
       // Get expiration or default to 24 hours
-      const expiration =
-        options?.expiration ||
-        new Date(Date.now() + 1000 * 60 * 60 * 24).toISOString();
+      expiration =
+        expiration || new Date(Date.now() + 1000 * 60 * 60 * 24).toISOString();
 
       // Prepare Sign in with Ethereum message
       const preparedMessage: Partial<SiweMessage> = {
-        domain: this.domain,
-        uri: this.origin,
-        address: ethers.utils.getAddress(address), // convert to EIP-55 format or else SIWE complains
+        domain: domain || 'localhost',
+        uri: origin || 'http://localhost',
+        address: ethers.utils.getAddress(address || signer?.address), // convert to EIP-55 format or else SIWE complains
         version: '1',
         chainId,
         expirationTime: expiration,
-        nonce: this.litNodeClient.latestBlockhash!,
+        nonce: litNodeClient.latestBlockhash!,
       };
 
       const message: SiweMessage = new SiweMessage(preparedMessage);
       const toSign: string = message.prepareMessage();
 
       // Use provided function to sign message
-      const signature = await signMessage(toSign);
+      const signature = await signer.signMessage(toSign);
 
       authSig = {
         sig: signature,
         derivedVia: 'web3.eth.personal.sign',
         signedMessage: toSign,
-        address: address,
+        address: signer.address,
       };
     } else {
-      if (!this.litNodeClient.latestBlockhash) {
+      if (!litNodeClient.latestBlockhash) {
```

Comment:
If we are going to throw an exception if the block hash isn't available we will want to check right when we enter this function as to not sign if we cannot get a `nonce` value.